Add GridMap function to change navigation map for baked navigation regions

smix8 opened the pull request:
Adds function to change the navigation map for baked navigation regions.
Before all cells with a baked navigation mesh were locked to the default navigation map of the world resource.

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.
-->

akien-mga reviewed (approved):
Looks good. Needs a rebase.

smix8 committed:
Add GridMap function to change navigation map for baked navigation regions

Adds function to change the navigation map for baked navigation regions.
Before all cells with a baked navigation mesh were locked to the default navigation map of the world resource.

akien-mga commented:
Thanks!
